### ISSUE TITLE
Respect user-set app scheme

### ISSUE THREAD

**Issue by sjchmiela** [simpleanalytics/expo-test-app#3]
When using Expo users can set a custom URL scheme to handle deep links etc. It may be nice to respect their choice in `url` property of the request sent to backend

https://github.com/simpleanalytics/expo-test-app/blob/61caba8e1753323fef72b9a1e9ca7ba2792b605e/react-native-simple-analytics.js#L34

Document on how one would create deep links to app can be found [here](https://docs.expo.io/versions/latest/workflow/linking/#expolinking-module), and API docs of the Linking module [here](https://docs.expo.io/versions/v34.0.0/sdk/linking/#linkingmakeurlpath-queryparams).

Basically you could probably use something like `Linking.makeUrl(namespace, { screenName })`.

**Comment by sjchmiela** [comment]
People may also want to pass in more params within the logged URL, like `navigationState` (in which one would hold things like `productId`, `title` etc.

Maybe even an API of the library could provide a way for users to customize the URL? I don't know how SimpleAnalytics later lets people inspect their analytics, but maybe they may want to pass in more information in the logged link?

**Comment by adriaanvanrossum** [comment]
> It may be nice to respect their choice in url property of the request sent to backend

Good suggestion. I agree for the `screenName` part, want to keep the `namespace` as the app name really is. Do you know if it's possible to have a hook on `Linking.makeUrl()` or read it out somehow and link it to the `screenName`?
